spec/models/storage_location_spec.rb

```diff
@@ -118,20 +118,34 @@
         it "gives informative errors" do
           storage_location = create(:storage_location, :with_items, item_quantity: 10, item: item, organization: @organization)
           expect do
-            storage_location.decrease_inventory(distribution_but_too_much.to_a).errors
+            storage_location.decrease_inventory(distribution_but_too_much.line_item_values).errors
           end.to raise_error(Errors::InsufficientAllotment)
         end
 
         it "does not change inventory quantities if there is an error" do
           storage_location = create(:storage_location, :with_items, item_quantity: 10, item: item, organization: @organization)
           starting_size = storage_location.size
           begin
-            storage_location.decrease_inventory(distribution.to_a)
+            storage_location.decrease_inventory(distribution.line_item_values)
           rescue Errors::InsufficientAllotment
           end
           storage_location.reload
           expect(storage_location.size).to eq(starting_size)
         end
+
+        context "when the item is does not exist" do
+          it "it raises insufficient allotment error if item does not exist" do
+            nonexistent_item = {item_id: nil, quantity: 1, name: "Nonexistent Item"}
+            storage_location = create(:storage_location, :with_items, item_quantity: 10, item: item, organization: @organization)
+            starting_size = storage_location.size
+            begin
+              storage_location.decrease_inventory([nonexistent_item])
+            rescue Errors::InsufficientAllotment
+            end
```

Comment:
Typically in RSpec you'll do a test like this as follows:

```ruby
expect { storage_location.decrease_inventory([nonexistent_item]) }
  .to raise_error(Errors::InsufficientAllotment)
```

Reply:
I think I was just copying it from above.